content/productivity/simultaneous-attack/plugin.js

```diff
@@ -0,0 +1,211 @@
+// Simultaneous attack
+//
+// attack thing with many thing at the same time
+
+var pg = df.getProcgenUtils();
+
+// removes all the child nodes of an element
+var removeAllChildNodes = (parent) => {
+    while (parent.firstChild) {
+        parent.removeChild(parent.firstChild);
+    }
+};
+
+// makes a (sometimes) clickable planet link 
+// kinda like https://github.com/darkforest-eth/client/blob/304f40a0b05dd111ce834d525c94ee0a5798e82f/src/Frontend/Components/Text.tsx#L237
+var planetLink = (locationId, clickable = true) => {
+    const planet = df.getPlanetWithId(locationId);
+    const planetElement = document.createElement(clickable ? "button" : "span");
+    planetElement.innerText = `L${planet.planetLevel}R${planet.upgradeState.reduce((a, b) => a + b, 0)} ${pg.getPlanetName(planet)}`;
+    planetElement.title = locationId;
+    planetElement.style.textDecoration = "underline";
+    planetElement.style.background = "none";
+    planetElement.style.border = "none";
+    planetElement.style.color = "white";
+    planetElement.style.outline = "none";
+    planetElement.style.padding = "0";
+    if (clickable) {
+        planetElement.addEventListener("click", () => {
```

Comment:
You addeventlistener quite often, but I dont see them being removed at all.

Reply:
Ok! I'll make the event listeners all methods of the class so I can cancel them. They should be removed on the destroy method, right?

Reply:
It seems like for these, they should be removed when you click the 'clear' button?

Reply:
The `clear` button is for clearing the attackers, not event listeners :) I should probably make that more clear!

Reply:
Oh I understand, but youre creating a listener for each attacker. Im not honestly sure its a problem, but you're clearing the list, you could clear the handlers too